src/base/DeviceAllocation.hh

```diff
@@ -40,15 +40,15 @@ class DeviceAllocation
     DeviceAllocation() = default;
 
     // Construct and allocate a number of bytes
-    DeviceAllocation(size_type num_bytes);
+    DeviceAllocation(std::size_t num_bytes);
 
     // Swap with another allocation
     inline void swap(DeviceAllocation& other) noexcept;
 
     //// ACCESSORS ////
 
     //! Get the number of bytes allocated
-    size_type size() const { return size_; }
+    std::size_t size() const { return size_; }
```

Comment:
I think we should locally typedef `size_type = std::size_t` but we might have to be careful about how it's used (anywhere in class scope is fine, but we would have to change  a hypothetical definition `size_type DeviceAllocation()::size()` to `auto ... -> size_type`).